gilded-rose-llc-frontend/pages/admin.tsx

```diff
@@ -3,35 +3,44 @@ import { GetServerSideProps } from 'next'
 import * as api from 'core/client'
 import Title from 'components/title'
 import { AdminItem } from 'core/admin-item'
+import React, { useState } from 'react'
 
 export const getServerSideProps: GetServerSideProps = async () => {
   const gateway = new api.ApiClient(fetch)
-  const items = await gateway.getAllItems()
-  return { props: { items } }
+  const initialItems = await gateway.getAllItems()
+  return { props: { initialItems } }
 }
 
 interface AdminProps {
-  items: AdminItem[]
+  initialItems: AdminItem[]
 }
 
-const Admin: NextPage<AdminProps> = ({ items }) => {
+const Admin: NextPage<AdminProps> = ({ initialItems }) => {
+  const [items, setItems] = useState(initialItems)
+  const updateItems = async () => {
+    const gateway = new api.ApiClient(fetch)
```

Comment:
Suggestion: since the `ApiClient` is used in multiple spots, consider exporting it as a singleton.

Reply:
Is this similar to what we worked on @elijahsullivan?

@mrfishball [React Context?](https://reactjs.org/docs/context.html)

Reply:
Yea I think the idea is similar in that we should only have one instance of the `ApiClient` and it can be shared among all pages/components easily. I think Context works even better than just having it being a Singleton class.